ExtractorUtils/ExtractorRunner.cs

```diff
@@ -79,34 +76,39 @@ public static class ExtractorRunner
                     services.AddSingleton<BaseExtractor>(prov => prov.GetRequiredService<TExtractor>());
                     DateTime startTime = DateTime.UtcNow;
                     ILogger<BaseExtractor> log;
-                    using (var provider = services.BuildServiceProvider())
+                    await using var provider = services.BuildServiceProvider();
+
+                    if (addMetrics)
                     {
-                        if (addMetrics)
-                        {
-                            var metrics = provider.GetRequiredService<MetricsService>();
-                        }
-                        log = new NullLogger<BaseExtractor>();
-                        if (addLogger)
-                        {
-                            log = provider.GetRequiredService<ILogger<BaseExtractor>>();
-                        }
-                        var extractor = provider.GetRequiredService<TExtractor>();
-                        if (onCreateExtractor != null)
-                        {
-                            var destination = provider.GetRequiredService<CogniteDestination>();
-                            onCreateExtractor(destination, extractor);
-                        }
-                        try
-                        {
-                            await extractor.Start(source.Token).ConfigureAwait(false);
-                        }
-                        catch (TaskCanceledException) when (source.IsCancellationRequested)
-                        {
-                            log.LogWarning("Extractor stopped manually");
-                        }
-                        catch (Exception ex)
+                        var metrics = provider.GetRequiredService<MetricsService>();
+                    }
+                    log = new NullLogger<BaseExtractor>();
+                    if (addLogger)
+                    {
+                        log = provider.GetRequiredService<ILogger<BaseExtractor>>();
+                    }
+                    var extractor = provider.GetRequiredService<TExtractor>();
+                    if (onCreateExtractor != null)
+                    {
+                        var destination = provider.GetRequiredService<CogniteDestination>();
+                        onCreateExtractor(destination, extractor);
+                    }
+                    var run = provider.GetService<ExtractionRun>();
+                    try
+                    {
+                        await extractor.Start(source.Token).ConfigureAwait(false);
+                    }
+                    catch (TaskCanceledException) when (source.IsCancellationRequested)
+                    {
+                        log.LogWarning("Extractor stopped manually");
```

Comment:
Where should successful runs be reported?

Reply:
By default they are reported on disposal of the Run object, so at the end of this block.

Reply:
They may also be reported in the `OnStop` method of BaseExtractor, or at any other point. The run object is designed so that you tell it when it should stop, or it stops automatically if disposed before finished.